src/app/business/common/components/searchBar/searchBar.js

```diff
@@ -0,0 +1,332 @@
+
+            // set item in redux, and launch related sagas for fetching list if needed
+            setState({
+                isParent: item.isLogic || !selectedParent,
+                inputValue: '',
+                selectedItem: newSelectedItem,
+                item: selectedParent || item.isLogic ? item.label : '',
+                toUpdate,
+            });
+        }
+    };
+
+    handleDelete = (item) => () => {
+        const {setState, selectedItem} = this.props;
+
+        // remove empty parent, the item we want to delete
+        let newSelectedItems = selectedItem.filter((o) => !(
+            (o.child === '' && !o.isLogic) // remove parent without child
+            || o.uuid === item.uuid), // remove item clicked
+
+        );
+
+        // remove -OR- item if not a chip before (i.e nothing or another -OR-)
+        newSelectedItems = newSelectedItems.filter((o, i) => !(
+            o.isLogic && i === 0 // remove first item if isLogic
+            || o.isLogic && i > 0 && newSelectedItems[i - 1].isLogic), // remove isLogic if precedent isLogic
+
+        );
+
+        // need to setItem correctly after deleting
+        const l = newSelectedItems.length,
+            last = l ? newSelectedItems[l - 1] : undefined;
+
+        setState({
+            selectedItem: newSelectedItems,
+            isParent: true,
+            item: last && last.isLogic ? '-OR-' : '',
+            toUpdate: true,
+        });
+
+        this.clickInput();
+    };
+
+    handleOuterClick = () => {
+        const {setState, inputValue} = this.props;
+
+        if (inputValue) {
+            setState({
+                inputValue: '',
+            });
+        }
+    };
+
+    clear = () => {
+        const {setState} = this.props;
+
+        setState({
+            selectedItem: [],
+            isParent: true,
+            item: '',
+            toUpdate: true,
+        });
+
+        this.clickInput();
+    };
+
+    clickInput = () => {
+        setTimeout(() => {
+            // should appear after blur of input (need to call setTimeout as downshift does)
+            // stay on focus
+            if (this.input.current) {
+                this.input.current.focus();
+            }
+        }, 0);
+    };
+
+    searchInput = (props) => {
+        const {suggestions, placeholder} = this.props;
+
+        // due to ugly render props philosophy of Downshift, we have to extract the declaration in a function...
+        return (
+            <SearchInputWrapper {...props.getRootProps({refKey: 'innerRef'})}>
+                <SearchInput
+                    handleKeyDown={this.handleKeyDown}
+                    handleInputChange={this.handleInputChange}
+                    handleDelete={this.handleDelete}
+                    suggestions={suggestions}
+                    clickInput={this.clickInput}
+                    input={this.input}
+                    placeholder={placeholder}
+                    key={this.input}
```

Comment:
Same question as above, this looks like an addition.

Reply:
Yes indeed, I got the same errors for this one.